runtime/common/RecordLogDecoder.cpp

```diff
@@ -51,41 +52,48 @@ void cudaq::RecordLogDecoder::decode(const std::string &outputLog) {
       break;
     case RecordType::END: {
       // indicates end of a shot
-      if (entries.size() < 2)
-        throw std::runtime_error("Missing shot status");
-      if ("0" != entries[1])
-        throw std::runtime_error("Cannot handle unsuccessful shot");
+      if (hasShotStatus) {
+        if (entries.size() < 2)
+          throw std::runtime_error("Missing shot status");
+        if ("0" != entries[1])
+          throw std::runtime_error("Cannot handle unsuccessful shot");
+      }
     } break;
     case RecordType::OUTPUT: {
-      if (entries.size() < 3)
-        throw std::runtime_error("Insufficent data in a record");
-      if ((schema == SchemaType::LABELED) && (entries.size() != 4))
-        throw std::runtime_error("Unexpected record size for a labeled record");
+      processOutputField(entries);
+    } break;
+    }
+  } // for line
+}
 
-      std::string recType = entries[1];
-      std::string recValue = entries[2];
-      std::string recLabel = (entries.size() == 4) ? entries[3] : "";
+void cudaq::RecordLogDecoder::processOutputField(
+    const std::vector<std::string> &entries) {
+  if (entries.size() < 3)
+    throw std::runtime_error("Insufficent data in a record");
+  if ((schema == SchemaType::LABELED) && (entries.size() != 4))
+    throw std::runtime_error("Unexpected record size for a labeled record");
 
-      if ("RESULT" == recType)
-        throw std::runtime_error("This type is not yet supported");
-      if ("TUPLE" == recType)
-        throw std::runtime_error("This type is not yet supported");
-      if ("ARRAY" == recType)
-        throw std::runtime_error("This type is not yet supported");
+  std::string recType = entries[1];
```

Comment:
```suggestion
  std::string &recType = entries[1];
```
Can these be references? To avoid a copy?